src/Akka.Persistence.Azure/TableEntities/AllPersistenceIdsEntry.cs

```diff
@@ -1,61 +1,58 @@
 ﻿using System;
-using System.Collections.Generic;
-using Microsoft.Azure.Cosmos.Table;
+using Azure;
+using Azure.Data.Tables;
 
 namespace Akka.Persistence.Azure.TableEntities
 {
-    internal sealed class AllPersistenceIdsEntry
-        : ITableEntity
+    internal sealed class AllPersistenceIdsEntry 
     {
         private const string ManifestKeyName = "manifest";
         public const string PartitionKeyValue = "allPersistenceIdsIdx";
 
         // In order to use this in a TableQuery a parameterless constructor is required
-        public AllPersistenceIdsEntry()
+        public AllPersistenceIdsEntry(TableEntity entity)
         {
+            PartitionKey = entity.PartitionKey;
+            ETag = entity.ETag;
+            RowKey = entity.RowKey;
+            Timestamp = entity.Timestamp;
+            
+            Manifest = entity.ContainsKey(ManifestKeyName)
+                ? entity.GetString(ManifestKeyName)
+                : string.Empty;
         }
 
         public AllPersistenceIdsEntry(
             string persistenceId,
             string manifest = "")
         {
             PartitionKey = PartitionKeyValue;
-
             RowKey = persistenceId;
-
             Manifest = manifest;
         }
 
-        public string ETag { get; set; }
-
-        public string Manifest { get; set; }
-
-        public string PartitionKey { get; set; }
+        public string PartitionKey { get; }
+        public ETag ETag { get; }
+        public string RowKey { get; }
+        public DateTimeOffset? Timestamp { get; }
 
-        public string RowKey { get; set; }
+        public string Manifest { get; }
 
-        public DateTimeOffset Timestamp { get; set; }
 
-        public void ReadEntity(
-            IDictionary<string, EntityProperty> properties,
-            OperationContext operationContext)
+        public TableEntity WriteEntity()
         {
-            Manifest =
-                properties.ContainsKey(ManifestKeyName)
-                    ? properties[ManifestKeyName].StringValue
-                    : string.Empty;
-        }
-
-        public IDictionary<string, EntityProperty> WriteEntity(
```

Comment:
The `ReadEntity` and `WriteEntity` de/serialization methods are removed from the new `ITableEntity` interface, the new interface does not have a custom serialization mechanism built in, instead it is relying on the "new" microsoft json serializer which might conflicts with newtonsoft serializer. Opted to not use it instead to safe the headache of trying to make it work using property attribute magics.

Reply:
Does this mean potential upgrade problems for users rolling forward?

Reply:
Ok, got it - heard your explanation over the call this morning.